runtime/src/startup.rs

```diff
@@ -52,14 +55,24 @@ macro_rules! stack_size {
 // from start, which is written directly in assembly.
 #[no_mangle]
 extern "C" fn rust_start() -> ! {
-    // TODO: Call memop() to inform the kernel of the stack and heap sizes +
-    // locations. Also, perhaps we should support calling a heap initialization
+    // TODO: Call memop() to inform the kernel of the stack and size +
+    // locations (for debugging).
+    // Also, perhaps we should support calling a heap initialization
     // function?
 
     extern "Rust" {
+        // This function is created by the set_main!() macro.
         fn libtock_unsafe_main() -> !;
     }
+    // TODO: Provide mechanism for dynamic heap size
+    let app_heap_size: usize = 1024;
```

Comment:
A heap size mechanism does not belong in `libtock_runtime`, that belongs in a separate crate, so this TODO is incorrect.

The heap size is 0 until the heap allocator (if one is present) decides otherwise. As such, app_heap_size should be 0 here.

Reply:
Do you know why the default heap size was 1024 in libtock1, regardless of whether the alloc feature was enabled (https://github.com/tock/libtock-rs/blob/022984f9adf1469ba7f53f8d4a7911ea08abe799/build.rs#L31)? That is why I took this approach here.

Reply:
No idea. It's very likely that was a leftover from before I made alloc optional in libtock1.

Reply:
That makes sense, seems that we should also default to 0 here.